src/platform/datapath_winuser.c

```diff
@@ -4491,6 +4528,16 @@ CxPlatSocketSendInline(
         CMsg->cmsg_len = WSA_CMSG_LEN(sizeof(INT));
         *(PINT)WSA_CMSG_DATA(CMsg) = SendData->ECN;
 
+        if (Socket->Datapath->Features & CXPLAT_DATAPATH_FEATURE_SEND_DSCP) {
+            WSAMhdr.Control.len += WSA_CMSG_SPACE(sizeof(INT));
+            CMsg = WSA_CMSG_NXTHDR(&WSAMhdr, CMsg);
+            CXPLAT_DBG_ASSERT(CMsg != NULL);
+            CMsg->cmsg_level = IPPROTO_IP;
+            CMsg->cmsg_type = IP_TOS;
+            CMsg->cmsg_len = WSA_CMSG_LEN(sizeof(INT));
+            *(PINT)WSA_CMSG_DATA(CMsg) = SendData->ECN | (SendData->DSCP << 2);
+        }
```

Comment:
This is something that came up before for ECN, but I think it might be worth doing now: Don't actually set the ancillary data at all, if the final result is zero, which would just match what the stack already puts in the packet. It just results in extra work for no net change in behavior....